### PR TITLE
Add content for the sequence diagram section

### PR DESCRIPTION
**Pull request by CharukaK**
$subject

### CODE REVIEW

File: en/docs/visual-programming/sequence-diagram-view.md [hunk 1]
```diff
@@ -1 +1,83 @@
-# Sequence diagram view
\ No newline at end of file
+# Sequence diagram view
+With Ballerina you can view your code in graphical mode and develop your program logic through it. This
```
Comment (praneesha) [inline en/docs/visual-programming/sequence-diagram-view.md:2]:
```suggestion
With Ballerina, you can view your code in a graphical mode and develop your program logic through it. This
```

File: en/docs/visual-programming/sequence-diagram-view.md [hunk 2]
```diff
@@ -1 +1,83 @@
...
+allows you to look at your program logic holistically and gives you the ability to comprehend the program at 
+a glance. This visual representation has its roots in sequence diagrams and can capture how the logic of
+the program flows, how concurrent execution flow works and what remote endpoints are involved and how they interact
```
Comment (praneesha) [inline en/docs/visual-programming/sequence-diagram-view.md:5]:
```suggestion
the program flows, how the concurrent execution flow works, which remote endpoints are involved, and how they interact 
```

File: en/docs/visual-programming/sequence-diagram-view.md [hunk 3]
```diff
@@ -1 +1,83 @@
...
\ No newline at end of file
+# Sequence diagram view
+With Ballerina you can view your code in graphical mode and develop your program logic through it. This
+allows you to look at your program logic holistically and gives you the ability to comprehend the program at 
+a glance. This visual representation has its roots in sequence diagrams and can capture how the logic of
+the program flows, how concurrent execution flow works and what remote endpoints are involved and how they interact 
+with the different objects in the program.
+
+## Switching to graphical mode
+There are two methods of switching to the graphical mode using the vs-code extension
```
Comment (praneesha) [inline en/docs/visual-programming/sequence-diagram-view.md:9]:
```suggestion
There are two methods for switching to the graphical mode using the VS Code extension.
```

File: en/docs/visual-programming/sequence-diagram-view.md [hunk 4]
```diff
@@ -1 +1,83 @@
...
+
+1. By executing the `Ballerina: Show Diagram` command from the command palette.
+    ![command screenshot](./img/show-diagram-command-palette.png)
+2. By using the `Show Diagram Button` in the editor group.
+    ![button screenshot](./img/editor-group-button.png)
+
+## Graphical Representation of code
```
Comment (praneesha) [inline en/docs/visual-programming/sequence-diagram-view.md:16]:
```suggestion
## Graphical representation of code
```

File: en/docs/visual-programming/sequence-diagram-view.md [hunk 5]
```diff
@@ -1 +1,83 @@
...
+
+### Functions
+Graphical representations of functions are based on sequence diagrams, each function has a lifeline with a start and
```
Comment (praneesha) [inline en/docs/visual-programming/sequence-diagram-view.md:19]:
```suggestion
Graphical representations of functions are based on sequence diagrams. Each function has a lifeline with a start and 
```

File: en/docs/visual-programming/sequence-diagram-view.md [hunk 6]
```diff
@@ -1 +1,83 @@
...
+1. By executing the `Ballerina: Show Diagram` command from the command palette.
+    ![command screenshot](./img/show-diagram-command-palette.png)
+2. By using the `Show Diagram Button` in the editor group.
+    ![button screenshot](./img/editor-group-button.png)
+
+## Graphical Representation of code
+
+### Functions
+Graphical representations of functions are based on sequence diagrams, each function has a lifeline with a start and 
+end, and this lifeline represents the flow of the function logic.
```
Comment (praneesha) [inline en/docs/visual-programming/sequence-diagram-view.md:20]:
```suggestion
end and this lifeline represent the flow of the function logic. 
```

File: en/docs/visual-programming/sequence-diagram-view.md [hunk 7]
```diff
@@ -1 +1,83 @@
...
+    ![button screenshot](./img/editor-group-button.png)
+
+## Graphical Representation of code
+
+### Functions
+Graphical representations of functions are based on sequence diagrams, each function has a lifeline with a start and 
+end, and this lifeline represents the flow of the function logic. 
+![button screenshot](./img/function-sequence-diagram.png)
+
+### Concurrent Execution
```
Comment (praneesha) [inline en/docs/visual-programming/sequence-diagram-view.md:23]:
```suggestion
### Concurrent execution
```

File: en/docs/visual-programming/sequence-diagram-view.md [hunk 8]
```diff
@@ -1 +1,83 @@
...
+Concurrent executions are supported in Ballerina through workers. Each worker has a lifeline that runs parallel to 
+the lifeline of the main function body, the diagram also captures asynchronous messaging between workers.
```
Comment (praneesha) [inline en/docs/visual-programming/sequence-diagram-view.md:25]:
```suggestion
the lifeline of the main function body. The diagram also captures asynchronous messaging between workers.
```

File: en/docs/visual-programming/sequence-diagram-view.md [hunk 9]
```diff
@@ -1 +1,83 @@
...
+        msg -> A;
+    }
+
+    wait A;
+    io:println("Woker A execution finished");
+}
+```
+![worker screenshot](./img/worker_sample.png)
+
+### Remote Endpoints and Clients
```
Comment (praneesha) [inline en/docs/visual-programming/sequence-diagram-view.md:61]:
```suggestion
### Remote endpoints and clients
```

File: en/docs/visual-programming/sequence-diagram-view.md [hunk 10]
```diff
@@ -1 +1,83 @@
...
+Ballerina has support to specify interactions with remote endpoints through its syntax, and this ability also
```
Comment (praneesha) [inline en/docs/visual-programming/sequence-diagram-view.md:62]:
```suggestion
Ballerina has support to specify interactions with remote endpoints through its syntax. This ability also 
```

File: en/docs/visual-programming/sequence-diagram-view.md [hunk 11]
```diff
@@ -1 +1,83 @@
...
+}
+```
+![worker screenshot](./img/worker_sample.png)
+
+### Remote Endpoints and Clients
+Ballerina has support to specify interactions with remote endpoints through its syntax, and this ability also 
+translates into the graphical representation giving each endpoint its lifeline to represent the messages that it 
+can receive.
+
+The following code shows an HTTP client that is used used to GET and POST requests to a remote endpoint.
```
Comment (praneesha) [inline en/docs/visual-programming/sequence-diagram-view.md:66]:
```suggestion
The following code shows an HTTP client that is used to the `GET` and `POST` requests sent to a remote endpoint.
```

File: en/docs/visual-programming/sequence-diagram-view.md [hunk 12]
```diff
@@ -1 +1,83 @@
...
+    http:Client lookupService = check new (lookupUrl);
+    http:Client reportService = check new (reportUrl);
+
+    json result = check lookupService->get("/query");
+    http:Response response = check reportService->post("/report", result);
+}
+```
+![graphical screenshot](./img/client-code-representations.png)
+
+## Developing programs in Visual Mode
```
Comment (praneesha) [inline en/docs/visual-programming/sequence-diagram-view.md:79]:
```suggestion
## Developing programs in the visual mode
```

File: en/docs/visual-programming/sequence-diagram-view.md [hunk 13]
```diff
@@ -1 +1,83 @@
...
+You can interact with the diagram using the `plus` icons placed in the lifeline, you can add a statement at the
```
Comment (praneesha) [inline en/docs/visual-programming/sequence-diagram-view.md:80]:
```suggestion
You can interact with the diagram using the `plus` icons placed in the lifeline. You can add a statement at the 
```

File: en/docs/visual-programming/sequence-diagram-view.md [hunk 14]
```diff
@@ -1 +1,83 @@
...
+
+    json result = check lookupService->get("/query");
+    http:Response response = check reportService->post("/report", result);
+}
+```
+![graphical screenshot](./img/client-code-representations.png)
+
+## Developing programs in Visual Mode
+You can interact with the diagram using the `plus` icons placed in the lifeline, you can add a statement at the 
+function execution flow by clicking on the add button.
```
Comment (praneesha) [inline en/docs/visual-programming/sequence-diagram-view.md:81]:
```suggestion
function execution flow by clicking the **Add** button.
```

File: en/docs/visual-programming/sequence-diagram-view.md [hunk 15]
```diff
@@ -1 +1,83 @@
...
+With Ballerina you can view your code in graphical mode and develop your program logic through it. This
+allows you to look at your program logic holistically and gives you the ability to comprehend the program at 
+a glance. This visual representation has its roots in sequence diagrams and can capture how the logic of
+the program flows, how concurrent execution flow works and what remote endpoints are involved and how they interact 
+with the different objects in the program.
+
+## Switching to graphical mode
+There are two methods of switching to the graphical mode using the vs-code extension
+
+1. By executing the `Ballerina: Show Diagram` command from the command palette.
```
Comment (nilminiwso2) [inline en/docs/visual-programming/sequence-diagram-view.md:11]:
```suggestion
- By executing the `Ballerina: Show Diagram` command from the command palette.
```

File: en/docs/visual-programming/sequence-diagram-view.md [hunk 16]
```diff
@@ -1 +1,83 @@
...
+    ![command screenshot](./img/show-diagram-command-palette.png)
+2. By using the `Show Diagram Button` in the editor group.
```
Comment (nilminiwso2) [inline en/docs/visual-programming/sequence-diagram-view.md:13]:
```suggestion
- Click the `Show Diagram` button in the editor group.
```

Comment (nilminiwso2) [reply en/docs/visual-programming/sequence-diagram-view.md:13]: what is the `editor group`?

File: en/docs/visual-programming/sequence-diagram-view.md [hunk 17]
```diff
@@ -1 +1,83 @@
...
+### Functions
+Graphical representations of functions are based on sequence diagrams, each function has a lifeline with a start and 
+end, and this lifeline represents the flow of the function logic. 
+![button screenshot](./img/function-sequence-diagram.png)
+
+### Concurrent Execution
+Concurrent executions are supported in Ballerina through workers. Each worker has a lifeline that runs parallel to 
+the lifeline of the main function body, the diagram also captures asynchronous messaging between workers.
+
+The following example shows a sample of messages passing between workers and waiting in the main function until the
```
Comment (nilminiwso2) [inline en/docs/visual-programming/sequence-diagram-view.md:27]: `sample of messaging passing..` or `list of messages passing..`?
@praneesha

File: en/docs/visual-programming/sequence-diagram-view.md (hunk 14 above)
Comment (nilminiwso2) [reply en/docs/visual-programming/sequence-diagram-view.md:81]: by clicking **Add**.
we don't use the word `button`

File: en/docs/visual-programming/sequence-diagram-view.md (hunk 17 above)
Comment (CharukaK) [reply en/docs/visual-programming/sequence-diagram-view.md]: Changed the description as 
`The following example shows how asynchronous messaging is represented in the diagram.`
in ee4df6e

File: en/docs/visual-programming/sequence-diagram-view.md [hunk 18]
```diff
@@ -1 +1,82 @@
...
+        // Sends the `Hello` string to the `A` worker asynchronously.
+        string msg = "Hello";
+        msg -> A;
+    }
+
+    wait A;
+    io:println("Woker A execution finished");
+}
+```
+![worker screenshot](./img/worker_sample.png)
```
Comment (nilminiwso2) [inline en/docs/visual-programming/sequence-diagram-view.md:58]: Is this the `docs/img` folder?
Can we move all images to the `docs/img` folder please? and update the links.

Comment (CharukaK) [reply en/docs/visual-programming/sequence-diagram-view.md]: Updated with [0d63093](https://github.com/wso2/docs-ballerina-visual-studio-code-plugin/pull/8/commits/0d63093b907cdbcbcba748b0fc98b4169a5bb974)

**Review by nilminiwso2 — APPROVED** [review]

**Pull request merged**